tests/server/dataSource-tests.js

```diff
@@ -0,0 +1,565 @@
+ * @return {Object} - a couchDB style view response.
+ */
+sjrk.test.storyTelling.server.dataSource.recordsToStoryViewResponse = function (records) {
+    var filter = ["title", "author", "tags", "content"];
+
+    var response = {
+        offset: 0,
+        rows: []
+    };
+
+    fluid.each(records, function (record, id) {
+        var story = fluid.copy(record.value);
+
+        if (story.published) {
+            story = fluid.filterKeys(story, filter);
+
+            response.rows.push({
+                "key": id,
+                "id": id,
+                "value": story
+            });
+        }
+    });
+    response.total_rows = response.rows.length;
+
+    return response;
+};
+
+fluid.defaults("sjrk.test.storyTelling.server.dataSource.testEnvironment", {
+    gradeNames: ["fluid.test.testEnvironment"],
+    components: {
+        testDB: {
+            type: "sjrk.test.storyTelling.server.mockDatabase",
+            createOnEvent: "{datasourceTester}.events.onTestCaseStart",
+            options: {
+                components: {
+                    authorsDBConfig: {
+                        options: {
+                            dbDocuments: sjrk.test.storyTelling.server.dataSource.mockRecords.authors
+                        }
+                    },
+                    storiesDBConfig: {
+                        options: {
+                            dbDocuments: sjrk.test.storyTelling.server.dataSource.mockRecords.stories
+                        }
+                    }
+                }
+            }
+        },
+        // a DataSource to get a list of stories
+        viewDataSource: {
+            type: "sjrk.storyTelling.server.dataSource.couch.view",
+            options: {
+                host: "http://localhost:6789"
+            }
+        },
+        // a DataSource to get a stories by author
+        storyByAuthorDataSource: {
+            type: "sjrk.storyTelling.server.dataSource.couch.authorStoriesView",
+            options: {
+                host: "http://localhost:6789"
+            }
+        },
+        // a DataSource to get or save a single story
+        storyDataSource: {
+            type: "sjrk.storyTelling.server.dataSource.couch.story",
+            options: {
+                host: "http://localhost:6789"
+            }
+        },
+        // a DataSource to delete a single story
+        deleteStoryDataSource: {
+            type: "sjrk.storyTelling.server.dataSource.couch.deleteStory",
+            options: {
+                host: "http://localhost:6789"
+            }
+        },
+        datasourceTester: {
+            type: "sjrk.test.storyTelling.server.dataSource.tester"
+        }
+    }
+});
+
+fluid.defaults("sjrk.test.storyTelling.server.dataSource.tester", {
+    gradeNames: ["fluid.test.testCaseHolder"],
+    members: {
+        rev: null
+    },
+    testOpts: {
+        storiesViewError: {
+            "reason": "ddoc _design/stories has no view named notAView",
```

Comment:
Is this meant to read "ddoc" or "doc"? If it's the former, it may be more clear if it says "design doc"

Reply:
that's an error straight from the database, it isn't something I defined.

Reply:
Curious! But okay, let's just leave it as-is, in that case